fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java

```diff
@@ -83,43 +83,43 @@ private void setupBackButton() {
   }
 
   public void setupNavigationBar() {
+    BottomNavigationView navbar = findViewById(R.id.lao_detail_nav_bar);
+    mViewModel.getCurrentTab().observe(this, tab -> navbar.setSelectedItemId(tab.getMenuId()));
     navbar.setOnItemSelectedListener(
         item -> {
-          int id = item.getItemId();
-          if (id == R.id.lao_detail_event_list_menu) {
-            openDetailMenu();
-          } else if (id == R.id.lao_detail_identity_menu) {
-            openIdentityMenu();
-          } else if (id == R.id.lao_detail_witnessing_menu) {
-            openWitnessMenu();
-          } else if (id == R.id.lao_detail_digital_cash_menu) {
-            openDigitalCashMenu();
-          } else if (id == R.id.lao_detail_social_media_menu) {
-            openSocialMediaMenu();
-          }
+          LaoTab tab = LaoTab.findByMenu(item.getItemId());
+          openTab(tab);
           return true;
         });
+    // Set an empty reselect listener to disable the onSelectListener when pressing multiple times
+    navbar.setOnItemReselectedListener(item -> {});
   }
 
-  private void openSocialMediaMenu() {
-    startActivity(
-        SocialMediaActivity.newIntent(
-            this,
-            mViewModel.getCurrentLaoValue().getId(),
-            mViewModel.getCurrentLaoValue().getName()));
-  }
-
-  private void openDigitalCashMenu() {
-    startActivity(
-        DigitalCashActivity.newIntent(
-            this,
-            mViewModel.getCurrentLaoValue().getId(),
-            mViewModel.getCurrentLaoValue().getName()));
+  private void openTab(LaoTab tab) {
+    switch (tab) {
+      case EVENTS:
+        openEventsMenu();
+        break;
+      case IDENTITY:
+        openIdentityMenu();
+        break;
+      case WITNESSING:
+        openWitnessMenu();
+        break;
+      case DIGITAL_CASH:
+        openDigitalCashMenu();
+        break;
+      case SOCIAL:
+        openSocialMediaMenu();
+        break;
+      default:
+        Log.w(TAG, "Unhandled tab type : " + tab);
+    }
   }
 
-  private void openWitnessMenu() {
+  private void openEventsMenu() {
     setCurrentFragment(
-        getSupportFragmentManager(), R.id.fragment_witnessing, WitnessingFragment::newInstance);
+        getSupportFragmentManager(), R.id.fragment_lao_detail, LaoDetailFragment::newInstance);
```

Comment:
Are you implicitly saying the `LaoDetailFragment` should be renamed LaoEventsFragment ? 😏

Reply:
It should

Reply:
I must have done that subconsciously. But it's a good proposition I made.